jellyfin-core/src/main/kotlin/org/jellyfin/sdk/discovery/RecommendedServerDiscovery.kt

```diff
@@ -27,103 +29,95 @@ public class RecommendedServerDiscovery(
+		val client = jellyfin.createApi(
+			baseUrl = address,
+			httpClientOptions = HttpClientOptions(
+				followRedirects = false,
+				timeout = HTTP_TIMEOUT
+			)
+		)
 		val api = SystemApi(client)
 
-		val startTime = System.currentTimeMillis()
-
-		@Suppress("TooGenericExceptionCaught")
-		val info = try {
-			api.getPublicSystemInfo()
-		} catch (err: ConnectException) {
-			logger.debug("Could not connect to $address", err)
-			null
-		} catch (err: Exception) {
-			logger.error("Could not retrieve public system info for $address", err)
-			null
+		val info: Response<PublicSystemInfo>?
+		val responseTime = measureTimeMillis {
+			@Suppress("TooGenericExceptionCaught")
+			info = try {
+				api.getPublicSystemInfo()
+			} catch (err: ConnectException) {
+				logger.debug("Could not connect to $address", err)
+				null
+			} catch (err: Throwable) {
+				logger.error("Could not retrieve public system info for $address", err)
+				null
+			}
 		}
-		val endTime = System.currentTimeMillis()
 
 		return SystemInfoResult(
 			address = address,
 			systemInfo = if (info != null && info.status == HTTP_OK) info.content else null,
-			responseTime = endTime - startTime,
+			responseTime = responseTime,
 		)
 	}
 
-	@Suppress("MagicNumber")
-	private fun assignScore(result: SystemInfoResult, parentResult: SystemInfoResult? = null): RecommendedServerInfo {
-		var points = 0
-
-		// Security
-		if (result.address.startsWith(HTTPS_PREFIX)) points += 3
-
-		// Speed
-		when {
-			result.responseTime < 500 -> points += 3
-			result.responseTime < 1500 -> points += 2
-			result.responseTime < 5000 -> points += 1
-		}
-
-		// Compatibility
-		val version = result.systemInfo?.version?.let(ServerVersion::fromString)
-		if (version != null) {
-			if (version >= Jellyfin.apiVersion) points += 1
-			if (version >= Jellyfin.minimumVersion) points += 1
-		}
-
-		val productName = result.systemInfo?.productName
-		if (productName != null && !productName.equals(PRODUCT_NAME, ignoreCase = true)) points = 0
-
-		// Minimum amount of points: 0
-		// Maximum amount of points: 8
-		val score = when {
-			points < 3 -> RecommendedServerInfoScore.BAD
-			points < 6 -> RecommendedServerInfoScore.OK
-			else -> RecommendedServerInfoScore.GOOD
-		}
-
-		return RecommendedServerInfo(result.address, result.responseTime, score, result.systemInfo, parentResult?.address)
+	/**
+	 * Discover all servers in the [servers] flow and retrieve the public system information to assign a score.
+	 * Returned servers are not necessarily in order of score. Use [minimumScore] to automatically remove bad matches.
+	 */
+	public suspend fun discover(
+		servers: Flow<String>,
+		minimumScore: RecommendedServerInfoScore,
+	): Flow<RecommendedServerInfo> = withContext(Dispatchers.IO) {
+		servers
+			.map(::getSystemInfoResult)
+			.mapNotNull(::assignScore)
+			.filter {
+				// Use [minimumScore] to filter out bad score matches
+				it.score.score >= minimumScore.score
```

Comment:
If `RecommendedServerInfoScore` was comparable, this could be shorter.

Reply:
Can't override `compareTo` from an enum.